pkg/policy/api/selector.go

```diff
@@ -56,6 +62,17 @@ func (n EndpointSelector) String() string {
 	return string(j)
 }
 
+// CachedString returns the cached string representation of the LabelSelector
+// within this EndpointSelector.
+func (n *EndpointSelector) CachedString() string {
+	if atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&n.cachedString))) == nil {
+		j := n.LabelSelector.String()
+		atomic.StorePointer(&n.cachedString, unsafe.Pointer(&j))
+		return j
+	}
+	return *(*string)(atomic.LoadPointer(&n.cachedString))
```

Comment:
Would be better to load the pointer only once in the beginning, the second load is not unnecessary, and can not be optimized away by the compiler, and could return nil, if the pointer is manipulated concurrently. So:
```
func (n *EndpointSelector) String() string {
	p := atomic.LoadPointer(&n.cachedString)
	if p == nil {
		j := n.LabelSelector.String()
		atomic.StorePointer(&n.cachedString, unsafe.Pointer(&j))
		return j
	}
	return *(*string)p
}
```